components/plurals/src/operands.rs

```diff
@@ -33,45 +33,46 @@ use fixed_decimal::FixedDecimal;
 ///
 /// ```
 /// use icu::plurals::PluralOperands;
-/// assert_eq!(PluralOperands {
-///    i: 2,
-///    v: 0,
-///    w: 0,
-///    f: 0,
-///    t: 0,
-///    c: 0,
-/// }, PluralOperands::from(2_usize))
+/// assert_eq!(PluralOperands::new(
+///    2, // i
+///    0, // v
+///    0, // w
+///    0, // f
+///    0, // t
+///    0, // c
+/// ), PluralOperands::from(2_usize))
 /// ```
 ///
 /// From &str
 ///
 /// ```
 /// use icu::plurals::PluralOperands;
-/// assert_eq!(Ok(PluralOperands {
-///    i: 123,
-///    v: 2,
-///    w: 2,
-///    f: 45,
-///    t: 45,
-///    c: 0,
-/// }), "123.45".parse())
+/// assert_eq!(Ok(PluralOperands::new(
+///    123, // i
+///    2, // v
+///    2, // w
+///    45, // f
+///    45, // t
+///    0, // c
+/// )), "123.45".parse())
 /// ```
 ///
 /// From [`FixedDecimal`]
 ///
 /// ```
 /// use fixed_decimal::FixedDecimal;
 /// use icu::plurals::PluralOperands;
-/// assert_eq!(Ok(PluralOperands {
-///    i: 123,
-///    v: 2,
-///    w: 2,
-///    f: 45,
-///    t: 45,
-///    c: 0,
-/// }), FixedDecimal::from(12345).multiplied_pow10(-2).map(|d| (&d).into()))
+/// assert_eq!(Ok(PluralOperands::new(
+///    123, // i
+///    2, // v
+///    2, // w
+///    45, // f
+///    45, // t
+///    0, // c
+/// )), FixedDecimal::from(12345).multiplied_pow10(-2).map(|d| (&d).into()))
 /// ```
-#[derive(Debug, Clone, Copy, PartialEq)]
+#[derive(Debug, Clone, Copy, PartialEq, Default)]
+#[non_exhaustive]
```

Comment:
cc @sffc @zbraniecki for discussing this choice

(from the previous thread: shane feels that this should be non_exhaustive, zibi feels that it should not)

Reply:
I'm torn on this. @eggrobin pointed out that if CLDR did decide to add an additional operand, we would have to figure out how to deal with data coming from CLDR with that operand, including how to gracefully fall back in patterns that contain the operand. Even if we held this back for an ICU4X major version bump, we still need to reason about generating data for older ICU4X versions for a certain period of time.